ingestion/src/metadata/ingestion/source/database/processor.py

```diff
@@ -0,0 +1,166 @@
+"""
+import logging
+import re
+from abc import ABC, abstractmethod
+from enum import Enum, auto
+from typing import Optional
+
+from commonregex import CommonRegex
+
+from metadata.generated.schema.api.data.createTable import CreateTableRequest
+from metadata.generated.schema.type.tagLabel import TagLabel
+from metadata.ingestion.api.processor import Processor, ProcessorStatus
+from metadata.ingestion.ometa.ometa_api import OpenMetadata, OpenMetadataConnection
+
+
+class PiiTypes(Enum):
+    """PiiTypes enumerates the different types of PII data"""
+
+    NONE = auto()
+    UNSUPPORTED = auto()
+    PHONE = auto()
+    EMAIL = auto()
+    CREDIT_CARD = auto()
+    ADDRESS = auto()
+    ADDRESS_LOCATION = auto()
+    PERSON = auto()
+    LOCATION = auto()
+    BIRTH_DATE = auto()
+    GENDER = auto()
+    NATIONALITY = auto()
+    IP_ADDRESS = auto()
+    SSN = auto()
+    USER_NAME = auto()
+    PASSWORD = auto()
+    ETHNICITY = auto()
+    TAX_ID = auto()
+    KEY = auto()
+
+
+class Scanner(ABC):
+    @abstractmethod
+    def scan(self, text):
+        """scan the text and return array of PiiTypes that are found"""
+
+
+class RegexScanner(Scanner):
+    """A scanner that uses commmon regular expressions to find PII"""
+
+    def scan(self, text):
+        """Scan the text and return an array of PiiTypes that are found"""
+        regex_result = CommonRegex(text)
+        types = []
+        if regex_result.phones:  # pylint: disable=no-member
+            types.append(PiiTypes.PHONE.name)
+        if regex_result.emails:  # pylint: disable=no-member
+            types.append(PiiTypes.EMAIL.name)
+        if regex_result.credit_cards:  # pylint: disable=no-member
+            types.append(PiiTypes.CREDIT_CARD.name)
+        if regex_result.street_addresses:  # pylint: disable=no-member
+            types.append(PiiTypes.ADDRESS.name)
+
+        return types
+
+
+class ColumnNameScanner(Scanner):
+    """
+    Column Name Scanner to scan column name
+    """
+
+    regex = {
+        PiiTypes.PERSON: re.compile(
+            "^.*(firstname|fname|lastname|lname|"
+            "fullname|maidenname|_name|"
+            "nickname|name_suffix|name).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.EMAIL: re.compile("^.*(email|e-mail|mail).*$", re.IGNORECASE),
+        PiiTypes.BIRTH_DATE: re.compile(
+            "^.*(date_of_birth|dateofbirth|dob|"
+            "birthday|date_of_death|dateofdeath).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.GENDER: re.compile("^.*(gender).*$", re.IGNORECASE),
+        PiiTypes.NATIONALITY: re.compile("^.*(nationality).*$", re.IGNORECASE),
+        PiiTypes.ADDRESS: re.compile(
+            "^.*(address|city|state|county|country|"
+            "zipcode|zip|postal|zone|borough).*$",
+            re.IGNORECASE,
+        ),
+        PiiTypes.USER_NAME: re.compile("^.*user(id|name|).*$", re.IGNORECASE),
+        PiiTypes.PASSWORD: re.compile("^.*pass.*$", re.IGNORECASE),
```

Comment:
I think we are catching a pattern that is too generic here. Columns with names such as `compassionating` or `bypass` for example would be tagged as PII

Reply:
Oh yes 👍